src/radicalpy/classical.py

```diff
@@ -19,6 +19,38 @@ def get_delta_r(mutual_diffusion: float, delta_T: float) -> float:
     return np.sqrt(6 * mutual_diffusion * delta_T)
 
 
+def kinetics(
+    time: np.ndarray, initial_populations: list, states: list, rate_equations: dict
+) -> np.ndarray:
+    """Kinetic rate equation solver.
+
+    Constructs the matrix propagator and performs a time evolution simulation.
+
+    Args:
+            time (np.ndarray): The timescale of the reaction kinetics (s).
+            initial_populations (list): The initial populations of all states.
```

Comment:
@lmantill  Could we combine `states` and `initial_populations`  into a `dict[str, float]`?

Reply:
okay